tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/BaseOnboard.kt

```diff
@@ -263,137 +274,183 @@ abstract class BaseOnboard : Runnable {
-            ).asJson()
-            .bodyOrThrow()
+            )
+        )
+
+        createRestClient(NetworkRestResource::class).use { client ->
+            client.start().proxy.setupHostedIdentities(holdingId, request)
+        }
     }
 
     protected fun register(waitForFinalStatus: Boolean = true) {
-        val response = Unirest.post("/membership/$holdingId")
-            .body(
-                mapOf(
-                    "memberRegistrationRequest" to mapOf(
-                        "context" to registrationContext
-                    )
-                )
-            ).asJson()
+        val registrationContext: Map<String, String> = registrationContext.mapValues { (_, value) ->
+            value.toString()
+        }
+
+        val request = MemberRegistrationRequest(
+            context = registrationContext
+        )
+
+        val response = createRestClient(MemberRegistrationRestResource::class).use { client ->
+            client.start().proxy.startRegistration(holdingId, request)
+        }
+
+        val registrationId = response.registrationId
+        val submissionStatus = response.registrationStatus
 
-        val body = response.bodyOrThrow().`object`
-        val submissionStatus = body.get("registrationStatus")
         if (submissionStatus != "SUBMITTED") {
-            throw OnboardException("Could not submit MGM registration: ${response.body}")
+            throw OnboardException("Could not submit MGM registration: ${response.memberInfoSubmitted}")
         }
-        val id = body.get("registrationId").toString()
-        println("Registration ID of $x500Name is $id")
+
+        println("Registration ID of $x500Name is $registrationId")
+
         if (waitForFinalStatus) {
-            waitForFinalStatus(id)
+            waitForFinalStatus(registrationId)
         }
     }
 
-    private fun waitForFinalStatus(id: String) {
-        val end = System.currentTimeMillis() + 5 * 60 * 1000
-        while (System.currentTimeMillis() < end) {
-            Thread.sleep(400)
-            val status = Unirest.get("/membership/$holdingId/$id").asJson()
-            val registrationStatus = status.bodyOrThrow().`object`.get("registrationStatus")
-            when (registrationStatus) {
-                "APPROVED" -> {
-                    return
-                }
-                "DECLINED", "INVALID", "FAILED" -> {
-                    throw OnboardException("Status of registration is $registrationStatus.")
-                }
-                else -> {
-                    println("Status of $x500Name registration is $registrationStatus")
+    private fun waitForFinalStatus(registrationId: String) {
+        createRestClient(MemberRegistrationRestResource::class).use { client ->
+            checkInvariant(
+                maxAttempts = MAX_ATTEMPTS,
+                waitInterval = WAIT_INTERVAL,
+                errorMessage = "Check Registration Progress: Invariant check failed after maximum attempts."
```

Comment:
```suggestion
                errorMessage = "Check Registration Progress failed after maximum number of attempts ($MAX_ATTEMPTS)."
```

Keep in mind invariant is an internal detail, so if you print that to the console it will just be confusing to the user that has no knowledge of the internal implementation.